server/src/main/java/com/paypal/selion/grid/servlets/ProxyInfo.java

```diff
@@ -0,0 +1,249 @@
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.JsonObject;
+
+import org.openqa.grid.common.RegistrationRequest;
+import org.openqa.grid.internal.RemoteProxy;
+import org.openqa.grid.internal.TestSlot;
+import org.openqa.selenium.remote.CapabilityType;
+
+import com.paypal.selion.node.servlets.LogServlet;
+import com.paypal.selion.proxy.SeLionRemoteProxy;
+
+@SuppressWarnings("unused")
+final class ProxyInfo {
+    /**
+     * Information not available
+     */
+    private static final String NOT_AVAILABLE = "not available";
+
+    /**
+     * Proxy is online
+     */
+    private static final String ONLINE = "online";
+
+    /**
+     * Proxy is offline
+     */
+    private static final String OFFLINE = "offline";
+
+    /**
+     * The url to view logs (via LogServlet) on the proxy. Defaults to {@value #NOT_AVAILABLE}
+     * 
+     * @see SeLionRemoteProxy#canViewLogs()
+     */
+    private String logsLocation;
+
+    /**
+     * @see RemoteProxy#isBusy()
+     */
+    private boolean isBusy;
+
+    /**
+     * @see RemoteProxy#getResourceUsageInPercent()
+     */
+    private float percentResourceUsage;
+
+    /**
+     * @see RemoteProxy#getTotalUsed()
+     */
+    private int totalUsed;
+
+    /**
+     * @see SeLionRemoteProxy#isScheduledForRecycle(). Defaults to false
+     */
+    private boolean isShuttingDown;
+
+    /**
+     * @see SeLionRemoteProxy#getTotalSessionsComplete(). Defaults to -1
+     */
+    private int totalSessionsComplete;
+
+    /**
+     * @see SeLionRemoteProxy#getTotalSessionsStarted(). Defaults to -1
+     */
+    private int totalSessionsStarted;
+
+    /**
+     * @see SeLionRemoteProxy#getUptimeInMinutes(). Defaults to -1
+     */
+    private long uptimeInMinutes;
+
+    /**
+     * @see RemoteProxy#getConfig()
+     */
+    private Map<String, Object> configuration;
+
+    /**
+     * Calls {@link RemoteProxy#getStatus()} to determine if proxy is {@value #ONLINE} or {@value #OFFLINE}.
+     */
+    private String status;
+
+    /**
+     * Calls {@link RemoteProxy#getStatus()} to determine proxy version. Defaults to {@value #NOT_AVAILABLE}
+     */
+    private String version;
+
+    /**
+     * Calls {@link RemoteProxy#getStatus()} to determine proxy OS. Defaults to {@value #NOT_AVAILABLE}
+     */
+    private String os;
```

Comment:
Why not initialize these to "NOT_AVAILABLE" since this is the for this and above fields.